compiler/src/main/java/com/dylibso/chicory/compiler/internal/WasmAnalyzer.java

```diff
@@ -67,7 +93,13 @@ public List<CompilerInstruction> analyze(int funcId) {
         // find label targets
         Set<Integer> labels = new HashSet<>();
 
-        for (AnnotatedInstruction ins : body.instructions()) {
+        HashMap<Integer, TryCatchBlock> tryCatchBlocks = new HashMap<>();
+        ArrayDeque<TryCatchBlock> tryCatchBlocksStack = new ArrayDeque<TryCatchBlock>();
+        HashMap<Integer, AnnotatedInstruction> tryTables = new HashMap<>();
```

Comment:
This is a little scary, do we really need 3 stateful data structure?
From the names is hard to guess what is used for what.